### PR TITLE
Revise the NPD task page

### PR DESCRIPTION
**Pull request by tengqm**
The NPD topic is pretty old. This PR revises the page with updated information.

### CODE REVIEW

File: content/en/docs/tasks/debug/debug-cluster/monitor-node-health.md [hunk 1]
```diff
@@ -100,54 +97,58 @@ This approach only applies to a Node Problem Detector started with `kubectl`.
 Overwriting a configuration is not supported if a Node Problem Detector runs as a cluster Addon.
 The Addon manager does not support `ConfigMap`.
 
-## Kernel Monitor
+## Problem Daemon
```
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/monitor-node-health.md:100]:
```suggestion
## Problem daemons
```
?

File: content/en/docs/tasks/debug/debug-cluster/monitor-node-health.md [hunk 2]
```diff
@@ -100,54 +97,58 @@ This approach only applies to a Node Problem Detector started with `kubectl`.
...
-configuration.
+- A `SystemStatsMonitor` type of daemon collects various health-related system stats as metrics.
+  You can customize its behavior by updating its
+  [configuration file](https://github.com/kubernetes/node-problem-detector/blob/master/config/system-stats-monitor.json).
 
-### Add new NodeConditions
+- A `CustomPluginMonitor` type of daemon invokes and checks various node problems by running
+  user-defined scripts. You can use different custom plugin monitors to monitor different
+  problems and customize the daemon behavior by updating the
+  [configuration file](https://github.com/kubernetes/node-problem-detector/blob/master/config/custom-plugin-monitor.json).
```
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/monitor-node-health.md:121]: I might link to a specific commit, which protects against https://github.com/kubernetes/node-problem-detector/ getting reorganized.

File: content/en/docs/tasks/debug/debug-cluster/monitor-node-health.md [hunk 3]
```diff
@@ -100,54 +97,58 @@ This approach only applies to a Node Problem Detector started with `kubectl`.
...
-  "type": "temporary/permanent",
-  "condition": "NodeConditionOfPermanentIssue",
-  "reason": "CamelCaseShortReason",
-  "message": "regexp matching the issue in the kernel log"
-}
-```
+You can extend the Node Problem Detector to execute any monitor scripts written in any language by
+developing a custom plugin. The monitor scripts must conform to the plugin protocol in exit code
+and standard output. For more information, please refer to the
+[plugin interface proposal](https://docs.google.com/document/d/1jK_5YloSYtboj-DtfjmYKxfNnUxCAvohLnsH5aGCAYQ/edit#).
```
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/monitor-node-health.md:136]: Due to the risk of vandalism, let's not link to that doc. We could ask SIG Node to make a tidied version where people with the link only have read access.

File: content/en/docs/tasks/debug/debug-cluster/monitor-node-health.md [hunk 4]
```diff
@@ -100,54 +97,58 @@ This approach only applies to a Node Problem Detector started with `kubectl`.
...
 
-### Add support for another log format {#support-other-log-format}
+- **Kubernetes exporter**: this exporter reports node problems to the Kubernetes API server.
+  Temporary problems are reported as Events and permanent problems are reported as Node Conditions.
 
-Kernel monitor uses the
-[`Translator`](https://github.com/kubernetes/node-problem-detector/blob/v0.1/pkg/kernelmonitor/translator/translator.go) plugin to translate the internal data structure of the kernel log.
-You can implement a new translator for a new log format.
+- **Prometheus exporter**: this exporter reports node problems and metrics locally as Prometheus
+  metrics. The connection to the Prometheus server can be specified using command line flags.
```
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/monitor-node-health.md:147]: Is this right?
```suggestion
  (or OpenMetrics) metrics. You can specify the IP address and port for the exporter using a
  command line argument.
```

File: content/en/docs/tasks/debug/debug-cluster/monitor-node-health.md [hunk 5]
```diff
@@ -12,7 +12,7 @@ weight: 20
 *Node Problem Detector* is a daemon for monitoring and reporting about a node's health.
 You can run Node Problem Detector as a `DaemonSet` or as a standalone daemon.
 Node Problem Detector collects information about node problems from various daemons
-and reports these conditions to the API server as [NodeCondition](/docs/concepts/architecture/nodes/#condition)
+and reports these conditions to the API server as [node condition](/docs/concepts/architecture/nodes/#condition)
 and [Event](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#event-v1-core).
```
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/monitor-node-health.md:16]:
```suggestion
or as [Events](/docs/reference/kubernetes-api/cluster-resources/event-v1/).
```
?

**Review by sftim — COMMENTED** [review]
Thanks.

A few thoughts.

Happy to pre-approve these changes; they are already a big improvement.

--- 

From the existing page:
> ```markdown
> You can also enable Node Problem Detector with `kubectl` or by creating an Addon pod.
> ```

Maybe change “pod” to “DaemonSet”?

File: content/en/docs/tasks/debug/debug-cluster/monitor-node-health.md (hunk 3 above)
Comment (tengqm) [reply content/en/docs/tasks/debug/debug-cluster/monitor-node-health.md:136]: Posted a comment to the google docs page asking for alternative links. I'm open to any improvement ideas.

File: content/en/docs/tasks/debug/debug-cluster/monitor-node-health.md (hunk 5 above)
Comment (tengqm) [reply content/en/docs/tasks/debug/debug-cluster/monitor-node-health.md:16]: OK

File: content/en/docs/tasks/debug/debug-cluster/monitor-node-health.md (hunk 2 above)
Comment (tengqm) [reply content/en/docs/tasks/debug/debug-cluster/monitor-node-health.md:121]: Pinned to 0.8.12 tag now.

**Comment by sftim** [comment]
/approve

If this gets /lgtm, I'm happy to see this merge.

**Pull request merged**